tests/ftests/systemd.py

```diff
@@ -27,3 +29,91 @@ def is_delegated(config, scope_name):
                 # until we figure out something better :(
                 return True
             raise re
+
+    # This function builds the absolute path of the cgroup, using the default
+    # systemd path and checks if the cgroup exists. The function can build
+    # path for all three cgroup setup modes, including empty cgroup v2.
+    @staticmethod
+    def cgroup_exists(config, ctrl_name, cgroup_name, ignore_systemd=False):
+        # 'None' is a valid controller name, while creating an empty cgroup
+        # on cgroup v2 is unified setup. But it doesn't have a mount point,
+        # lets use the first available controller mount point to form the path
+        if ctrl_name is None:
+            if ignore_systemd:
+                raise IndexError('Unsupported controller \'None\'')
+
+            controllers = ['cpu', 'memory', 'io', 'pids', 'cpuset']
+            for controller in controllers:
+                ctrl_mnt = Cgroup.get_controller_mount_point(controller)
+                if 'Unknown version' not in ctrl_mnt:
+                    break
+            else:
+                raise IndexError('No cgroup controllers found')
+        else:
+            ctrl_mnt = Cgroup.get_controller_mount_point(ctrl_name)
+
+        if (ignore_systemd):
+            cgrp_path = os.path.join(ctrl_mnt, cgroup_name)
+            return os.path.isdir(cgrp_path)
+
+        try:
+            with open('/run/libcgroup/systemd', 'r') as default_systemd_path:
+                slice_scope_path = default_systemd_path.readlines()[0]
+        except IOError:
+            return False
+
+        cgrp_path = os.path.join(ctrl_mnt, slice_scope_path, cgroup_name)
+        return os.path.isdir(cgrp_path)
+
+    # This is a simple wrapper, that calls Cgroup.create() to create the
+    # cgroup and cgroup_exists(), which builds the cgroup path and returns
+    # True if the cgroup exists, False otherwise
+    @staticmethod
+    def create_and_validate(config, ctrl_name, cgroup_name, ignore_systemd=False):
```

Comment:
`create()` is located in the Cgroup class.  Would it make sense to have `create_and_validate()` next to it in the same class?  (It seems like we should consider converting some of the `create()` calls in other tests to `create_and_validate()` and thus putting this in Cgroup class might make more sense.